doc/USAGE.md

```diff
@@ -0,0 +1,53 @@
+# pick_ik : Usage
+
+## Using pick_ik as a Kinematics Plugin
+
+As discussed in the MoveIt2 documentation, you can use the [MoveIt Setup Assistant](https://moveit.picknik.ai/humble/doc/examples/setup_assistant/setup_assistant_tutorial.html) or change the [`kinematics.yaml` file](https://moveit.picknik.ai/main/doc/examples/kinematics_configuration/kinematics_configuration_tutorial.html?highlight=kinematics%20yaml#kinematics-configuration) for your robot setup to use `pick_ik` as the IK solver.
+
+An example `kinematics.yaml` file might look as follows:
+
+```yaml
+panda_arm:
+  kinematics_solver: pick_ik/PickIkPlugin
+  kinematics_solver_timeout: 0.05
+  kinematics_solver_attempts: 3
+  mode: global
+  rotation_scale: 0.5
+  twist_threshold: 0.001
+  cost_threshold: 0.001
+  minimal_displacement_weight: 0.0
+  gd_step_size: 0.0001
+```
+
+As a sanity check, you could follow the [MoveIt Quickstart in RViz](https://moveit.picknik.ai/humble/doc/tutorials/quickstart_in_rviz/quickstart_in_rviz_tutorial.html) tutorial and change the `moveit_resources/panda_moveit_config/config/kinematics.yaml/kinematics.yaml` file to use a configuration like the one above.
+
+---
+
+## Parameter Description
+
+For an exhaustive list of parameters, refer to the [parameters YAML file](../src/pick_ik_parameters.yaml).
+
+Some key parameters you may want to start with are:
+
+* `mode`: If you choose `local`, this solver will only do local gradient descent; If you choose `global`, it will also enable the evolutionary algorithm. Using the global solver will be less performant, but if you're having trouble getting out of local minima, this could help you. We recommend using `local` for things like relative motion / Cartesian interpolation / endpoint jogging, and `global` if you need to solve for goals with a far-away initial conditions.
+* `memetic_<property>`: All the properties that only kick in if you use the `global` solver. The key one is `memetic_num_threads`, as we have enabled the evolutionary algorithm to solve on multiple threads.
+* `cost_threshold`: This solver works by setting up cost functions based on how far away your pose is, how much your joints move relative to the initial guess, and custom cost functions you can add. Optimization succeeds only if the cost is less than `cost_threshold`. Note that if you're adding custom cost functions, you may want to set this threshold fairly high and rely on `twist_threshold` to be your deciding factor whereas this is more of a guideline.
```

Comment:
Probably off-topic to this PR, but I would somewhat prefer slightly different parameter names.
* `_threshold` actually could be replaced with `*_limit` or a `max_*`
* twist implies velocity afaik, a simple `cartesian_distance_limit` or `pose_distance_limit` would fit better
* `rotation_scale` should be `rotation_weight` if it directly applies to the orientation cost
* If all of these apply to costs, adding a cost like `rotation_cost_weight` wouldn't hurt

These don't have to be addressed right now, but if we want to support possible more default cost types in the future, we should talk about how to make these parameters more consistent and maintainable in general.

Reply:
I've captured these in https://github.com/PickNikRobotics/pick_ik/issues/20 -- thanks!